test/parallel_api/numeric/numeric.ops/in_place_scan.pass.cpp

```diff
@@ -19,64 +19,570 @@
+    }
+
+    // specialization for non-random_access iterators
+    template <typename Policy, typename Iterator1, typename Size>
+    typename ::std::enable_if<!is_base_of_iterator_category<::std::random_access_iterator_tag, Iterator1>::value, void>::type
+    operator()(Policy&& exec, Iterator1 keys_first, Iterator1 keys_last, Size n)
+    {
+    }
+};
+
+DEFINE_TEST_1(test_exclusive_scan_inplace, BinaryOperation)
+{
+    DEFINE_TEST_CONSTRUCTOR(test_exclusive_scan_inplace)
+
+    const int kInitValue = 2;
+
+    // TODO: replace data generation with random data and update check to compare result to
+    // the result of a serial implementation of the algorithm
+    template <typename Iterator1, typename Size>
+    void
+    initialize_data(Iterator1 host_keys, Size n)
+    {
+        const Size kStartVal = 1;
+        
+        for (Size i = 0; i != n; ++i)
+            host_keys[i] = kStartVal + i;
+    }
+
+    // Required to pass source_host_keys_state param by copy for safe state of this param on caller side
+    template <typename ValT, typename Iterator1, typename Size,
+              typename BinaryOperationCheck = oneapi::dpl::__internal::__pstl_plus>
+    void check_values(std::vector<ValT> source_host_keys_state, Iterator1 host_keys, Size n,
+                      BinaryOperationCheck op = BinaryOperationCheck())
+    {
+        const auto source_host_keys_state_copy = source_host_keys_state;
+
+        if (source_host_keys_state.empty())
+            return;
+
+        exclusive_scan_serial(source_host_keys_state.begin(), source_host_keys_state.end(), source_host_keys_state.begin(), kInitValue, op);
+
+        EXPECT_EQ_N(source_host_keys_state.cbegin(), host_keys, source_host_keys_state.size(),
+                    "wrong effect from exclusive_scan_serial");
+    }
+
+#if TEST_DPCPP_BACKEND_PRESENT
+    // specialization for hetero policy
+    template <typename Policy, typename Iterator1, typename Size>
+    typename ::std::enable_if<
+        oneapi::dpl::__internal::__is_hetero_execution_policy<typename ::std::decay<Policy>::type>::value &&
+            is_base_of_iterator_category<::std::random_access_iterator_tag, Iterator1>::value,
+        void>::type
+    operator()(Policy&& exec, Iterator1 keys_first, Iterator1 keys_last,
+               Size n)
+    {
+        TestDataTransfer<UDTKind::eKeys, Size> host_keys(*this, n);
+
+        typedef typename ::std::iterator_traits<Iterator1>::value_type KeyT;
+
+        // call algorithm with no optional arguments
+        initialize_data(host_keys.get(), n);
+        update_data(host_keys);
+
+        // copy source host keys state into temp buf
+        typedef typename ::std::iterator_traits<Iterator1>::value_type ValT;
+        const std::vector<ValT> source_host_keys_state(host_keys.get(), host_keys.get() + n);
+
+        auto new_policy = make_new_policy<new_kernel_name<Policy, 0>>(exec);
+        if (n == 1636)
```

Comment:
What does "if-else" mean here?
1) What's a "magic" constant 1636?
2) "if" line and "else" line are absolutely the same.

Reply:
t was for old debug purposes.